lib/common/acl.c

```diff
@@ -290,15 +290,6 @@ pcmk__apply_acl(xmlNode *xml)
     }
 
     p = xml->_private;
-    if (is_not_set(p->flags, xpf_acl_read)
-        && is_not_set(p->flags, xpf_acl_write)) {
-
-        p->flags |= xpf_acl_deny;
-        p = xml->doc->_private;
-        crm_info("Applied default deny ACL for user '%s' to <%s>",
```

Comment:
We may still want to keep this message or something like it. We're no longer **applying a default ACL**, but it's implicitly applied.

Reply:
I don't actually remember whether I considered that at the time or not.

This approach in pcmk__apply_acl seemed to mesh well with the way pcmk__check_acl is already set up. pcmk_check_acl walks up the parent tree looking for an explicit denial. If it doesn't find one, it enforces an implicit denial.

Walking up the parents requires little bit of additional code -- nothing too complicated, I think walk up until we find some ACL set or until we reach the root of the tree. Whereas we can achieve the same result with a bit less code (and a more natural feeling IMO) this way: applying an ACL only if the user configured one, and then denying by default later in processing if no ACL is set.

Reply:
As a side note, I just noticed that the old version logs "Applied default deny" even when there's an explicit deny ACL set.